GVFS/GVFS.Platform.Mac/MacPlatform.cs

```diff
@@ -32,36 +36,62 @@ public override string GetDataRootForGVFS()
         public override string GetDataRootForGVFSComponent(string componentName)
         {
             return MacPlatform.GetDataRootForGVFSComponentImplementation(componentName);
-        }
-
+        }
+
         public override bool TryGetGVFSEnlistmentRoot(string directory, out string enlistmentRoot, out string errorMessage)
         {
             return MacPlatform.TryGetGVFSEnlistmentRootImplementation(directory, out enlistmentRoot, out errorMessage);
-        }
-
+        }
+
         public override string GetNamedPipeName(string enlistmentRoot)
         {
             return MacPlatform.GetNamedPipeNameImplementation(enlistmentRoot);
         }
-
-        public override FileBasedLock CreateFileBasedLock(
-            PhysicalFileSystem fileSystem,
-            ITracer tracer,
+
+        public override FileBasedLock CreateFileBasedLock(
+            PhysicalFileSystem fileSystem,
+            ITracer tracer,
             string lockPath)
         {
             return new MacFileBasedLock(fileSystem, tracer, lockPath);
-        }
-
+        }
+
+        public override Dictionary<string, string> GetPhysicalDiskInfo(string path, bool sizeStatsOnly)
+        {
+            // DiskUtil will return disk statistics in xml format
+            ProcessResult processResult = ProcessHelper.Run("diskutil", "info -plist /", true);

```

Comment:
Note this opens the door for us also recording things like volume size, size used, remaining, etc..

@mjcheetham of all the fancy ways to get the info from diskutil, I think parsing the xml seems like the cleanest.

Reply:
The XDocument streaming APIs are very fast and reliable from my experience.
(so long as you take precautions around schema validation etc)